### ISSUE TITLE
Some potential CLI improvements

### ISSUE THREAD

**Issue by t9999clint** [ublue-os/fleek#208]
**Is your feature request related to a problem? Please describe.**
I use a lot of different package managers as I'm always switching between different distros of linux to get my work done.
The thing that kept me away from NIX for the most part was not having a simple way to manage it's installed applications. Fleek does a lot to improve upon this.
However there's still a few suggestions I'd like to make about the syntax of fleek to make it match the behaviour or dnf or apt a little more closely.

**Describe the solution you'd like**
Aliases for the commands `fleek install`, `fleek uninstall` and `fleek upgrade` to the commands `fleek add`, `fleek remove` and `fleek update -a`

Not sure if it's possible, but if you were to add a command `fleek list` that would list all packages and their versions currently installed by nix, that would be super cool. I've tried `nix-env -q` but it doesn't show anything.

**Describe alternatives you've considered**
Right now I'm just using the commands that are already availible to me, so nothing here is of high priority. But it'd be nice to be able to use nix just like I do flatpak, apt, dnf, etc...
for listing nix stuff I've been using `ls -o ~/.nix-profile/bin/` which is pretty ugly

**Comment by t9999clint** [comment]
just noticed that `fleek show` already does the same as I wanted `fleek list` to do.
Only thing that's missing is the version numbers from fleek show.

**Comment by mtalexan** [comment]
Getting the versions of packages is more difficult than you think.  Nix isn't designed around specific versions of packages, it's designed around reproducible builds.  As a convention the `nixpkgs` does usually use released versions of packages for the source code and much of the configuration for those builds, but not everything/everyone does.  

In `nixpkgs` there's a convention that the name of a package is `{pkg_name}-{pkg_version}` when possible, but that isn't always possible.  That seems to be how the `nixpkgs` search site (https://search.nixos.org/packages) parses the versions to list.  But if you look at the actual derivation definition, or the `meta` for the package, you'll see there is no specific version number exposed in the output otherwise.

So yes there's often an input version for the derivation definition on `nixpkgs` since it gets used in URLs for getting source code and whatnot, and makes it easier to update to a newer version, but the output derivation itself only includes a version number sometimes and only by convention of `nixpkgs`, not necessarily elsewhere.

**Comment by mtalexan** [comment]
If you want to see the info `fleek` has to work from on this, try examining the metadata yourself.  You can run `nix eval --json 'nixpkgs#git.meta' | jq '.'` to get a pretty-printed JSON of the fields `git` from `nixpkgs` provides in its metadata.

**Comment by t9999clint** [comment]
Yeah I noticed how weird the versioning in nix packages are, that's why I was mentioned that I wasn't even sure it was possible in the original post.
Stuff here is really just a few suggestions to make things easier for normal linux users not used to nix; if the version listing is gonna be a huge pain to implement, than it's probably not worth doing.
